CHANGELOG.rst

```diff
@@ -2,6 +2,229 @@
+this need entirely in the future.
+
+For the ZGW API's, this even means you don't have to worry anymore of updating the
+configuration when you publish a new version of a "zaaktype" - the right version will
+now automatically be selected.
+
+
+
+Detailed changes
+----------------
+
+
+This contains the changes from the alpha, beta and fixes applied between the beta and
+stable version.
+
+
+**New features**
+
+* [:backend:`4267`, :backend:`4567`, :backend:`4577`] Improved the UX of the Objects
+  API registration options:
+
+    - Configuration is now in a modal and changes in configuration require an explicit
+      confirmation, meaning you can now explore more without potentially breaking the
+      configuration.
+    - Upgraded the API group, object type and object type version dropdowns with search
+      functionality.
+    - Configuration fields are now logically grouped. Optional settings are shown in a
+      collapsed group to declutter the UI.
+    - You can now select a catalogue from a dropdown (with search functionality) that
+      contains the document types to use.
+    - API groups (admin): you can now specify a catalogue and the descriptions of
+      document types to use rather than entering the API URL to a specific version.
+
+  These UX and configuration improvements are still work-in-progress, more will become
+  available in next releases and we will also rework the ZGW API registration options.
+* [:backend:`4051`] Added a better JSON-editor in a number of places, bringing them up
+  to parity with the editor in the form builder:
+
+    - Editing JSON logic triggers.
+    - Editing JSON logic variable assignment expressions.
+    - Editing service fetch mapping expressions.
+    - Viewing the JSON-definition of logic rules and/or actions.
+* [:backend:`4555`] Improved the UX of pre-fill configuration on the variables tab:
+
+    - There is now a single summary column for the prefill configuration, instead of
+      three separate columns.
+    - Improved the wording/language used to differentiate between authorizee/authorised
+      roles.
+    - Editing the configuration is now done in a separate modal.
+
+* [:backend:`4456`] The admin interface now clearly displays which environment you are
+  on. You can disable displaying this information, and you can change the text and
+  colors to easily differentiate between acceptance/production environments.
+* [:backend:`4488`] The submisson report PDF now no longer opens in a new tab/window,
+  the browser is forced to download it.
+* Support pre-filling form fields from existing data in the Objects API:
+
+    - [:backend:`4397`] Added ability to store an object reference on the submission so
+      that the information can be retrieve and pre-filled.
+    - [:backend:`4395`] Added a flag to specify if an existing object needs to be
+      updated during registration, or a new record should be created.
+
+  This feature is currently under heavy development.
```

Comment:
I'll add a feature flag to disable it. However, I can guarantee you already that for next release this will not be in the changelog since it's generated by `git log` and I really don't want to track extra things like "which changelog items should only be mentioned in the next release".